In [1]:
import pandas as pd
from IPython.display import display

# Import core modules directly; bypassing the CLI
from src.engine import SwingScreenerEngine
from src.models import RunOptions

In [2]:
# Define a small, focused universe for the demo
demo_symbols = ["TCS.NS", "RELIANCE.NS", "HDFCBANK.NS", "INFY.NS", "ITC.NS"]

# Instantiate the options exactly as the CLI parser
options = RunOptions(
    symbols=demo_symbols,
    cacheMode="auto",      
    quiet=True,            
    exportFile="demoOutput.xlsx"
)

In [3]:
# Initialize and run the engine
engine = SwingScreenerEngine()
resultsDf = engine.run(options)

displayColumns = [
    'symbol',
    'isFinancial',
    'sector',
    'grade',
    'totalScore',
    'profitabilityScore',
    'balanceSheetScore',
    'valuationScore',
    'flags',
    'peRatio', 
    'evToEbitda', 
    'netDebtToEbitda', 
    'flags'
]

# Display a compact, readable top-10 table
display(resultsDf[displayColumns].head(10))


🔍 Screening 5 stocks through Buffett/Lynch/Graham rules...


------------------------------------------------------------
  🟢 Strong Buy: 0  |  🟢 Buy: 0  |  🟡 Watchlist: 0
  Total screened: 5  |  Avg score: 51.8
------------------------------------------------------------

📊 Exported to demoOutput.xlsx

JSON exported to demoOutput_20260427_193727.json


,symbol,isFinancial,sector,grade,totalScore,profitabilityScore,balanceSheetScore,valuationScore,flags,peRatio,evToEbitda,netDebtToEbitda,flags
Rank,,,,,,,,,,,,,
1,TCS.NS,False,Technology,🟠 Hold,57.8,21.6,17.3,5.1,,17.936720,11.507,-0.488895,
2,ITC.NS,False,Consumer Defensive,🟠 Hold,56.3,19.7,18.7,4.4,,18.711561,13.331,-0.674208,
3,HDFCBANK.NS,True,Financial Services,🟡 Watchlist,52.9,10.0,15.0,13.4,,17.629993,NaN,NaN,
4,INFY.NS,False,Technology,🟡 Watchlist,51.8,19.7,16.0,2.9,,15.713524,1054.633,-0.618564,
5,RELIANCE.NS,False,Energy,🟡 Watchlist,40.1,3.8,10.7,11.8,,22.353138,11.922,0.863889,


## A Note on Data Anomalies

Two rows in this output illustrate real-world data caveats rather than code bugs. HDFCBANK.NS shows NaN for EV/EBITDA and Net Debt/EBITDA because these multiples are not structurally compatible with financial institutions: banks do not operate on traditional EBITDA and their balance sheets are deposit-driven, so standard enterprise valuation ratios can break. INFY.NS shows an EV/EBITDA outlier (1054.633), which is likely a yfinance data-quality issue (for example stale or inconsistent denominator inputs) rather than an economically meaningful valuation multiple. This is exactly why the screener uses a multi-factor composite framework with scoring guardrails instead of relying on any single raw metric; despite the outlier, INFY.NS still carries a reasonable valuationScore (~2.9), indicating the composite system is absorbing noisy data as intended.